src/main/java/org/opensearch/agent/tools/PPLTool.java

```diff
@@ -83,12 +87,47 @@ public class PPLTool implements Tool {
 
     private String contextPrompt;
 
+    private PPLModelType pplModelType;
+
     private static Gson gson = new Gson();
 
-    public PPLTool(Client client, String modelId, String contextPrompt) {
+    private static Map<String, String> defaultPromptDict;
+
+    static {
+        try {
+            defaultPromptDict = loadDefaultPromptDict();
+        } catch (IOException e) {
+            log.error("fail to load default prompt dict" + e.getMessage());
+            defaultPromptDict = new HashMap<>();
+        }
+    }
+
+    private enum PPLModelType {
+        CLAUDE,
+        FINETUNE;
+
+        public static PPLModelType from(String value) {
+            if (value.isEmpty()) {
+                return PPLModelType.CLAUDE;
+            }
+            try {
+                return PPLModelType.valueOf(value.toUpperCase(Locale.ROOT));
+            } catch (Exception e) {
+                throw new IllegalArgumentException("Wrong PPL Model type, should be CLAUDE or FINETUNE");
```

Comment:
Is this possible to break the initialization flow?

Reply:
Yes, if the customer provide the wrong model type. Do you think it's better to set the default model type claude always?

Reply:
Yes, let's do that.

Reply:
Already add the logic.